File: 1816-truncate-sentence/1816-truncate-sentence.c
char * truncateSentence(char * s, int k){
    int i = 0;
    int c = 0;
    
    while(s[i])
    {
        if(s[i] == ' ')
            ++c;
        if(c == k)
        {
            s[i] = '\0';
            --i;
        }
        ++i;
    }
    return s;
}

File: 1816-truncate-sentence/1816-truncate-sentence.swift
class Solution {
    func truncateSentence(_ s: String, _ k: Int) -> String {
        var res = [Character]()
        var cnt = 0
        
        for i in s {
            if i == " " {
                cnt += 1
                if cnt == k {
                    return String(res)
                }
            }
            res.append(i)
        }
        
        return String(res)
    }
}